modules/misunderstood/src/backend/db.ts

```diff
@@ -92,6 +92,16 @@ export default class Db {
       .select('*')
       .then((data: DbFlaggedEvent[]) => (data && data.length ? data[0] : null))
 
+    const parentEvent = await this.knex(EVENTS_TABLE_NAME)
+      .where({ botId, incomingEventId: event.eventId, direction: 'incoming' })
+      .select('id', 'threadId', 'sessionId', 'event')
+      .limit(1)
+      .first()
+
+    if (!parentEvent) {
+      return
+    }
+
     const { threadId, sessionId, id: messageId, event: eventDetails } = await this.knex(EVENTS_TABLE_NAME)
       .where({ botId, incomingEventId: event.eventId, direction: 'incoming' })
       .select('id', 'threadId', 'sessionId', 'event')
```

Comment:
Aren't you just duplicating the query done right here ?

Reply:
You're right, fixed in a465772